pyabc/epsilon/temperature.py

```diff
@@ -0,0 +1,502 @@
+
+        # addressing the std, not the var
+        eps_base = np.sqrt(temp_base)
+
+        if not self.k:
+            # initial iteration
+            self.k[t - 1] = eps_base
+
+        k_base = self.k[t - 1]
+
+        if acceptance_rate < self.min_rate:
+            logger.debug("DalyScheme: Reacting to low acceptance rate.")
+            # reduce reduction
+            k_base = self.alpha * k_base
+
+        self.k[t] = min(k_base, self.alpha * eps_base)
+        eps = eps_base - self.k[t]
+        temperature = eps**2
+
+        return temperature
+
+
+class FrielPettittScheme(TemperatureScheme):
+    """
+    Basically takes linear steps in log-space. See [#vyshemirsky2008]_.
+
+    .. [#vyshemirsky2008] Vyshemirsky, Vladislav, and Mark A. Girolami.
+        "Bayesian ranking of biochemical system models."
+        Bioinformatics 24.6 (2007): 833-839.
+    """
+
+    def __call__(self,
+                 t: int,
+                 get_weighted_distances: Callable,
+                 max_nr_populations: int,
+                 pdf_norm: float,
+                 kernel_scale: str,
+                 prev_temperature: float,
+                 acceptance_rate: float):
+        # needs a starting temperature
+        # if not available, return infinite temperature
+        if prev_temperature is None:
+            return np.inf
+
+        # check if we can compute a decay step
+        if max_nr_populations == np.inf:
+            raise ValueError("Can only perform FrielPettittScheme step with a "
+                             "finite max_nr_populations.")
+
+        # base temperature
+        temp_base = prev_temperature
+        beta_base = 1. / temp_base
+
+        # time to go
+        t_to_go = max_nr_populations - t
+
+        beta = beta_base + ((1. - beta_base) * 1 / t_to_go) ** 2
+
+        temperature = 1. / beta
+        return temperature
+
+
+class EssScheme(TemperatureScheme):
+    """
+    Try to keep the effective sample size constant.
+
+    Parameters
+    ----------
+    target_relative_ess: float
+        Targe relative effective sample size.
+    """
+
+    def __init__(self, target_relative_ess: float = 0.8):
+        self.target_relative_ess = target_relative_ess
+
+    def __call__(self,
+                 t: int,
+                 get_weighted_distances: Callable,
+                 max_nr_populations: int,
+                 pdf_norm: float,
+                 kernel_scale: str,
+                 prev_temperature: float,
+                 acceptance_rate: float):
+        # execute functino (expensive if in calibration)
+        df = get_weighted_distances()
+
+        weights = np.array(df['w'], dtype=float)
+        pdfs = np.array(df['distance'], dtype=float)
+
+        # compute rescaled posterior densities
+        if kernel_scale == SCALE_LIN:
+            values = pdfs / pdf_norm
+        else:  # kernel_scale == SCALE_LOG
+            values = np.exp(pdfs - pdf_norm)
+
+        # to pmf
```

Comment:
what is "pmf" ?

Reply:
probability mass function (i.e. normalized to sum 1)

Reply:
can you specify? I think I havent seen this abbreviation in the code before.. (if it is already present somewhere, then just keep it)

Reply:
will do